osgar/drivers/logzeromq.py

```diff
@@ -0,0 +1,57 @@
+"""
+  Logger for ZeroMQ communication
+"""
+
+from threading import Thread
+
+import zmq
+
+from osgar.bus import BusShutdownException
+
+
+class LogZeroMQ:
+    def __init__(self, config, bus):
+        mode = config['mode']
+        endpoint = config['endpoint']
+
+        context = zmq.Context()
+        print("Connecting to hello world server ...")
+        if mode == 'PULL':
+            self.socket = context.socket(zmq.PULL)
+            self.thread = Thread(target=self.run_input, daemon=True)
+        elif mode == 'PUSH':
+            self.socket = context.socket(zmq.PUSH)
+            self.thread = Thread(target=self.run_output, daemon=True)
```

Comment:
Would it be possible not to make those threads `daemon`? Shutting down daemon threads is abrupt and can leak resources as not even context managers are respected - the thread is killed on an OS level AFAIK. It seems both threads are well-behaved and exit when they are told to via bus mechanism so daemoning should not be needed.

Reply:
Well, so far my experience is not very good - without daemon it is even with current master (`osgar.go`) problematic to stop/kill ... but I should probably write this somewhere else??

Reply:
As we chatted, can we have daemon=False here now?

Reply:
I forgot that from "hacking" to get tests working - changed back to False